src/components/UserInputComponent/UserInputForm.js

```diff
@@ -259,15 +279,35 @@ function UserInputForm(props) {
                   size="large"
                   sx={{ minWidth: "1" }}
                   disabled={
-                    !(formik.isValid && formik.dirty) ||
-                    selectedWayareaWithNoGDOGID
+                    selectedCalculationType == "" ||
+                    (selectedCalculationType == "navigationline" &&
+                      (formik.values.vaylat == null ||
+                        formik.values.vaylat == "" ||
+                        selectedWayareaWithNoGDOGID)) ||
+                    (selectedCalculationType == "reittiviiva" &&
+                      formik.values.routename == "") ||
+                    (selectedCalculationType == "compare" &&
+                      (formik.values.routename == "" ||
+                        formik.values.vaylat == null ||
+                        formik.values.vaylat == "" ||
+                        selectedWayareaWithNoGDOGID))
```

Comment:
In conjunction with the above mentioned refactor, make this it's own function to validate

Reply:
done